### PR TITLE
jaxmodels

### PR DESCRIPTION
**Pull request by WSDeWitt**
**To do:** fill in this draft PR with details

- new module: `multidms/jaxmodels`
- example notebook: `notebooks/jaxmodels`

### CODE REVIEW

**Comment by jgallowa07** [comment]
This is great, thanks @WSDeWitt! I'm digesting the math, code and reproducing the analysis now. The next step will be a comparative analysis which will likely spur thoughts on the integration of `jaxmodels` to the rest of the `multidms` namespace. More to come!

**Comment by jgallowa07** [comment]
We can discuss these in today's meeting but I'll be posting some questions/discussion points here:

1. Use of $\alpha$ parameter: I'm not sure I fully grasp this. You state: "where $\alpha$ is the effective number of generations of selection". But is the number of generations not always 1 i.e the pre-selection generation, and post selection generation? AFAIU, log$\alpha$ is a [free parameter](https://github.com/matsengrp/multidms/blob/b82d83cb7905b6953a0de537d1ac548a9c22b8fc/multidms/jaxmodels.py#L131) we fit to scale the sigmoid - what does the sigmoid scaling have to do with this "generations" description (my popgen brain is firing but not quite clicking).
2. I'm unclear, still, what the motivation/advantage of using the counts-based model is. Is it so we can use negative binomial as opposed to a huber regression style loss? This may be reason enough as you explained about the dispersion param, but I could use a little more explanation on why we think this is better than the huber loss regression model.
3. `sklearn` dependency should be added to the [pyproject.toml](https://github.com/matsengrp/multidms/blob/main/pyproject.toml) If we're going to keep this. It seems this is only used for the warm up of the individual linear models.
4. To be clear, here, it seems like you must put the wildtype as the first variant of each condition, but remove it from the training data? I get that all functional scores are relative to the wildtype, but then how is the model being informed about the correct $\beta_0$ value to infer?
5. It seems the fitting procedure is not deterministic. I assume this has something to do with [scikit-learn.linear_model.Ridge.fit](https://scikit-learn.org/0.16/modules/generated/sklearn.linear_model.Ridge.html#sklearn.linear_model.Ridge.fit) needing a seed to be set? That said, the results are very similar between runs which is good
6. The distribution of beta's seems a little odd ... why are there some extremely large, positive values? Could this be the effect of the negative binomial dispersion parameter? This is also causing there to be some very large 
7. The code seems to be _very_ memory intensive. Without formal benchmarking, it seems that fitting just one of these models takes nearly 10G ram ... running 6 model fits in parallel crashed my computer's applications. This could be because I cannot share the training binarymaps between models - which was a nice feature of the current `multidms`. 

More to come

**Comment by jgallowa07** [comment]
To Try (after discussions with Will):

## Investigate large beta's:

- [ ] Try no warm start with scikit. 
- [ ] Filtering counts more strictly. 
- [ ] hyper param L2 regularization.
- [ ] Try lowering the tolerance
- [ ] Stops vs. non-stops
- [ ] Make sure that wildtype are first, as that is an assumption of the new code.

## Comparative analysis to old results
 
 TODO
